src/Mvc/Mvc/src/MvcServiceCollectionExtensions.cs

```diff
@@ -30,69 +31,156 @@ public static IMvcBuilder AddMvc(this IServiceCollection services)
 
-            AddDefaultFrameworkParts(builder.PartManager);
+            if (setupAction == null)
+            {
+                throw new ArgumentNullException(nameof(setupAction));
+            }
 
-            // Order added affects options setup order
+            var builder = services.AddMvc();
+            builder.Services.Configure(setupAction);
 
-            // Default framework order
-            builder.AddFormatterMappings();
-            builder.AddViews();
-            builder.AddRazorViewEngine();
-            builder.AddRazorPages();
-            builder.AddCacheTagHelper();
+            return builder;
+        }
 
-            // +1 order
-            builder.AddDataAnnotations(); // +1 order
+        /// <summary>
+        /// Adds services for controllers to the specified <see cref="IServiceCollection"/>. This method will not
+        /// register services used for views or pages.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <returns>An <see cref="IMvcBuilder"/> that can be used to further configure the MVC services.</returns>
+        /// <remarks>
+        /// <para>
+        /// This method configures the MVC services for the commonly used features with controllers for an API. This
+        /// combines the effects of <see cref="MvcCoreServiceCollectionExtensions.AddMvcCore(IServiceCollection)"/>,
+        /// <see cref="MvcApiExplorerMvcCoreBuilderExtensions.AddApiExplorer(IMvcCoreBuilder)"/>,
+        /// <see cref="MvcCoreMvcCoreBuilderExtensions.AddAuthorization(IMvcCoreBuilder)"/>,
+        /// <see cref="MvcCorsMvcCoreBuilderExtensions.AddCors(IMvcCoreBuilder)"/>,
+        /// <see cref="MvcDataAnnotationsMvcCoreBuilderExtensions.AddDataAnnotations(IMvcCoreBuilder)"/>,
+        /// and <see cref="MvcCoreMvcCoreBuilderExtensions.AddFormatterMappings(IMvcCoreBuilder)"/>.
+        /// </para>
+        /// <para>
+        /// To add services for controllers with views call <see cref="MvcViewFeaturesMvcCoreBuilderExtensions.AddViews(IMvcCoreBuilder)"/>
+        /// on the resulting builder.
+        /// </para>
+        /// <para>
+        /// To add services for pages call <see cref="MvcRazorPagesMvcCoreBuilderExtensions.AddRazorPages(IMvcCoreBuilder)"/>
+        /// on the resulting builder.
+        /// </para>
+        /// </remarks>
+        public static IMvcBuilder AddControllers(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
 
-            builder.AddCors();
+            // This method excludes all of the view-related services by default.
+            var builder = services
+                .AddMvcCore()
+                .AddApiExplorer()
+                .AddAuthorization()
+                .AddCors()
+                .AddDataAnnotations()
+                .AddFormatterMappings();
```

Comment:
These are all of the things that you'll get by default. 
- API explorer doesn't do anything unless you call its APIs
- Auth is based on attributes
- Cors is based on attributes
- Data annotations is based on attributes
- Format filter is based on attributes

All of them are pay for play. 

I debated leaving out format filter because no one uses it. But I think it's hard to explain why we left it out that it is to just put it in. I still find it super duper vexxxing that format filter requires a gesture.

Reply:
I'm not sure. I think it's up to them to decide if this is what they wanted all along.